### PR TITLE
Fixes #23

### LINKED ISSUE #23: Unit test for EsStream fails
The stream extension for elasticsearch is ignoring the results of the initial query to open the scroll.  Therefore the unit test to compare the results always fails.  I have a solution.

### PR DESCRIPTION
**Pull request by pbrannigan**
Thank you for this contribution.  I was using it for a project I was working on.  I updated the elasticsearch extension so that the unit tests would pass.  I also corrected a typo in the name of the unit test.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -118,7 +118,7 @@ lazy val elasticsearch = project.in(file("extensions/elasticsearch"))
     name := "akka-stream-extensions-elasticsearch",
     libraryDependencies ++= Seq(
       "com.typesafe.akka" %% "akka-stream" % "2.4.2",
-      "org.elasticsearch" % "elasticsearch" % "1.3.2"
+      "org.elasticsearch" % "elasticsearch" % "2.4.0"
```
Comment (klpx) [inline build.sbt]: Hi! I think it would be better to not include elastic to build. This is achieved by adding `% "provided,test"` to dependency expression

**Comment by klpx** [comment]
Hi! Just did the same things in my project when found that underlying elastic API is too old :)
And found this pr before start mine.
Just wanted to say: with synchronous elastic client this is really nice solution with scala Stream collection! Took note!

Bad thing that this PR is stuck so long time.

File: extensions/elasticsearch/src/main/scala/elasticsearchExtensions.scala [hunk 2]
```diff
@@ -27,36 +31,68 @@ trait EsStream {
...
-          if (docs.isEmpty) None -> None
-          else Some(docs) -> Some(nextSearchResp)
-        }
-      }
+      if (docs.isEmpty) Stream.Empty
+      else docs #:: next(newRes)
     }
-    .mapConcat(identity)
+
+    Source(req(query)).mapConcat { identity }
```
Comment (klpx) [inline extensions/elasticsearch/src/main/scala/elasticsearchExtensions.scala]: Initial response started immediatelly. If Source declared via queryAsStream much earlier (scrollKeepAlive) than Materialization (runWith(Sink)), than cursor may outdate.

Correct code is like this `Source.fromIterator(() => req(query).toIterator).mapConcat(identity)`

**Comment by Timshel** [comment]
Sorry, I will look at this next week.

**Comment by Timshel** [comment]
I merged your PR then realized that `Source.unfoldAsync` should work.

**Comment by pbrannigan** [comment]
Nice.  That is probably a better solution.  Thank you for following up on this.

**Pull request merged**